pym/portage/__init__.py

```diff
@@ -561,11 +561,20 @@ def create_trees(config_root=None, target_root=None, trees=None, env=None,
 				clean_env[k] = v
 		if depcachedir is not None:
 			clean_env['PORTAGE_DEPCACHEDIR'] = depcachedir
-		settings = config(config_root=None, target_root="/",
-			env=clean_env, eprefix=None)
-		settings.lock()
-		trees._running_eroot = settings['EROOT']
-		myroots.append((settings['EROOT'], settings))
+		bsettings = config(config_root=None, target_root="/",
+			env=clean_env, sysroot="/", eprefix=None)
+		bsettings.lock()
+		trees._running_eroot = bsettings['EROOT']
+		myroots.append((bsettings['EROOT'], bsettings))
+
+		if settings['SYSROOT'] == '/':
+			dsettings = config(clone=bsettings)
+			dsettings.lock()
+			myroots.append((dsettings['EROOT'], dsettings))
+		elif settings['SYSROOT'] == settings['ROOT']:
+			dsettings = config(clone=settings)
+			dsettings.lock()
+			myroots.append((dsettings['EROOT'], dsettings))
```

Comment:
Thanks to this assertion, it seems to me that we only need 2 config instances (just like before SYSROOT):
```python
		if self.sysroot != "/" and self.sysroot != self.target_root:
			writemsg(_("!!! Error: SYSROOT (currently %s) must "
				"equal / or ROOT (currently %s).\n") %
				(self.sysroot, self.target_root),
				noiselevel=-1)
			raise InvalidLocation(self.sysroot)
```
Maybe this is all we need:
```diff
diff --git a/pym/portage/__init__.py b/pym/portage/__init__.py
index 137351a69..4f3159524 100644
--- a/pym/portage/__init__.py
+++ b/pym/portage/__init__.py
@@ -562,7 +562,7 @@ def create_trees(config_root=None, target_root=None, trees=None, env=None,
 		if depcachedir is not None:
 			clean_env['PORTAGE_DEPCACHEDIR'] = depcachedir
 		settings = config(config_root=None, target_root="/",
-			env=clean_env, eprefix=None)
+			env=clean_env, sysroot="/", eprefix=None)
 		settings.lock()
 		trees._running_eroot = settings['EROOT']
 		myroots.append((settings['EROOT'], settings))
```